src/scheduler.rs

```diff
@@ -168,12 +168,22 @@ async fn fetch_idle(ctx: &Context, connection: &mut Imap, folder: Config) -> Int
                 return connection.fake_idle(ctx, Some(watch_folder)).await;
             }
 
-            // Mark expired messages for deletion.
-            if let Err(err) = delete_expired_imap_messages(ctx)
-                .await
-                .context("delete_expired_imap_messages failed")
-            {
-                warn!(ctx, "{:#}", err);
+            if folder == Config::ConfiguredInboxFolder {
+                // Mark expired messages for deletion.
+                if let Err(err) = delete_expired_imap_messages(ctx)
+                    .await
+                    .context("delete_expired_imap_messages failed")
+                {
+                    warn!(ctx, "{:#}", err);
+                }
+
+                if let Err(err) = connection
+                    .store_seen_flags(ctx)
+                    .await
+                    .context("store_seen_flags failed")
+                {
+                    warn!(ctx, "{:#}", err);
+                }
```

Comment:
This delays receiving the message (we have just been woken up with NEWDATA from IDLE) even more than before.  

Can we not as a first step do this expiration/markseen work after we have fetched all new messages?

Reply:
This PR is not worse than the current state: previously jobs marking messages as seen were also executed before fetching messages, see the `inbox_loop` code. If anything, this PR will reduce the delay, because marking all the MDNs you have received while being offline is now done with a single batched request.

See also the commit message (copied to PR description). I would like to move this after fetching, but currently we don't check that UIDNEXT has not advanced when we select Inbox and before going to IDLE. So for now we need to make sure we don't reselect folder between fetching and going IDLE. This can be fixed as a follow-up, but is outside of the scope of this PR.